stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/composer/MessageComposerViewModel.kt

```diff
@@ -0,0 +1,146 @@
+package io.getstream.chat.android.ui.message.composer
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import io.getstream.chat.android.client.models.Attachment
+import io.getstream.chat.android.client.models.Message
+import io.getstream.chat.android.common.composer.MessageComposerController
+import io.getstream.chat.android.common.state.Edit
+import io.getstream.chat.android.common.state.MessageAction
+import io.getstream.chat.android.common.state.MessageInputState
+import io.getstream.chat.android.common.state.MessageMode
+import io.getstream.chat.android.common.state.Reply
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
+
+/**
+ * ViewModel responsible for handling the composing and sending of messages.
+ *
+ * It relays all its core actions to a shared data source, as a central place for all the Composer logic.
+ * Additionally, all the core data that can be reused across our SDKs is available through shared data sources, while
+ * implementation-specific data is stored in respective in the [ViewModel].
+ *
+ * @param messageComposerController The controller used to relay all the actions and fetch all the state.
+ */
+public class MessageComposerViewModel(
+    private val messageComposerController: MessageComposerController,
+) : ViewModel() {
+
+    private val _messageInputState: MutableStateFlow<MessageInputState> = MutableStateFlow(MessageInputState())
+
+    /**
+     * UI state of the message input component.
+     */
+    public val messageInputState: StateFlow<MessageInputState> = _messageInputState
```

Comment:
I'd like to move this to `MessageComposerController` in a separate PR as this will potentially touch compose module